src/sonic-eventd/src/eventutils.cpp

```diff
@@ -0,0 +1,103 @@
+#include "eventutils.h"
+#include <string.h>
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <unistd.h>
+#include <nlohmann/json.hpp>
+#include "logger.h"
+#include "table.h"
+
+using namespace swss;
+using namespace std;
+
+using json = nlohmann::json;
+
+bool isValidSeverity(string severityStr) {
+    transform(severityStr.begin(), severityStr.end(), severityStr.begin(), ::toupper);
+    if (severityStr == EVENT_SEVERITY_MAJOR_STR) return true;
+    if (severityStr == EVENT_SEVERITY_CRITICAL_STR) return true;
+    if (severityStr == EVENT_SEVERITY_MINOR_STR) return true;
+    if (severityStr == EVENT_SEVERITY_WARNING_STR) return true;
+    if (severityStr == EVENT_SEVERITY_INFORMATIONAL_STR) return true;
+    return false;
+}
+
+bool isValidEnable(string enableStr) {
+    if (enableStr == EVENT_ENABLE_TRUE_STR) return true;
+    if (enableStr == EVENT_ENABLE_FALSE_STR) return true;
+    return false;
+}
+
+bool parse_config(const char *filename, unsigned int& days, unsigned int& count) {
+    days = EHT_MAX_DAYS;
+    count = EHT_MAX_ELEMS;
+    std::ifstream ifs(filename);
+
+    if (!ifs.is_open()) {
+        std::cerr << "Failed to open file: " << filename << std::endl;
+        return false;
+    }
+
+    json j;
+    try {
+      j = json::parse(ifs);
+    }
+    catch (const std::exception &e) {
```

Comment:
>exception [](http://example.com/codeflow?start=22&length=9)

Could you use more specific exception class?

Reply:
Updated to handle specific exception